New documentation content (#671)
graknlabs/docs: 001-typedb-old/02-fundamentals/02-database.md

---
pageTitle: Database
keywords: database
longTailKeywords: typedb database
summary: TypeDB database concept.
toc: false
---

<!--- 
Logically separate data, cannot touch each other (eg. stored in separately directories) & therefore copyable independently etc.,, not optimised for many (1-10s), database contains schema + data
-->

# Database

A database is the outermost container for data in a TypeDB knowledge graph. Like a relational database, it is commonly 
known to be a good practice to create a single database per application, but it is absolutely fine to create as many 
databases as your application needs. As a rule of thumb, it is recommended to start off with one database and create 
more if the requirement arises.

<div class="note">
[Note]

> TypeDB optimised for lesser number of databases. So the best practise would be to keep no more than 10 databases on 

Reviewer: ```suggestion
TypeDB optimized for a small number of databases, so best practises is to keep no more than 10 databases on 
```

Author: How about this version?
```suggestion
TypeDB is optimized for a smaller number of databases, so the best practice is to keep no more than 10 databases on 
```
